torch_geometric/nn/aggr/scaler.py

```diff
@@ -0,0 +1,74 @@
+from typing import Any, Dict, List, Optional, Union
+
+import torch
+from torch import Tensor
+
+from torch_geometric.nn.aggr import Aggregation, MultiAggregation
+from torch_geometric.utils import degree
+
+
+class DegreeScalerAggregation(Aggregation):
+    """
+    Class that combines together one or more aggregators and then transforms
+    the result with one or more scalers. The scalers are normalised by the
+    in-degree of the training set and so must be provided at construction.
+
+    Args:
+        aggr (string or list or Aggregation, optional): The list of
+            aggregations given as :class:`~torch_geometric.nn.aggr.Aggregation`
+            (or any string that automatically resolves to it).
+        scalers (list of str): Set of scaling function identifiers, namely
+                :obj:`"identity"`, :obj:`"amplification"`,
+                :obj:`"attenuation"`, :obj:`"linear"` and
+                :obj:`"inverse_linear"`.
+        deg (Tensor): Histogram of in-degrees of nodes in the training set,
+            used by scalers to normalize.
+        aggr_kwargs (List[Dict[str, Any]], optional): Arguments passed to the
+            respective aggregation functions in case it gets automatically
+            resolved. (default: :obj:`None`)
+    """
+    def __init__(self, aggrs: List[Union[Aggregation, str]],
+                 scalers: List[str], deg: Tensor,
+                 aggrs_kwargs: Optional[List[Dict[str, Any]]] = None):
+
+        super().__init__()
+
+        self.agg = MultiAggregation(aggrs, aggrs_kwargs)
```

Comment:
I think this is incorrect. We would only do this in case of lists of aggregations. Maybe worth it to out-source this as well (since it is shared logic with `MessagePassing`)

Reply:
Sorry I didn't follow this one - what part is incorrect?

Reply:
The input to aggregation is not necessarily a list of aggregations, so `MultiAggregation` would
complain.

Reply:
Ah I see, actually the type annotation is `aggrs: List[Union[Aggregation, str]],` but the docstring makes this confusing. Will add a check for this.